### ISSUE TITLE
[Feature request?] Atlas in composer doesn't render all vector features

### ISSUE THREAD

**Issue by ChameleonScales** [geometalab/Vector-Tiles-Reader-QGIS-Plugin#278]
When I try to render an atlas of various places with the QGIS composer, not all vector features get rendered. It seems only those that have been loaded in the canvas do.
Furthermore, the zoom value in the canvas affects what types of features get rendered too, even though the Atlas has a fixed zoom value.
It would make sense that vector layers loaded with this plugin behaved the same way other layers do in that regard.

I assume that's a feature to be added rather than a bug. Either way, does that seem solvable?

**Comment by mnboos** [comment]
The plugin keeps only the visible data in the source, the others are discarded. That is by design.

I am not familiar with the Atlas tool but it is, to my best knowledge, also by design, that vector maps control feature visibility based on zoom values.

> It would make sense that vector layers loaded with this plugin behaved the same way other layers do in that regard.

By that you mean that all the data is actually loaded/available?

**Comment by ChameleonScales** [comment]
No, what I mean is that when the atlas generates an image for a specific area and a specific zoom level, it should load the data for that area and that zoom level instead of using the data of the canvas which may be of a completely different area and zoom level.
This is already what it does with WMS, WMTS and XYZ feeds, but not with Vector Tiles Reader.
